### DISCUSSION TITLE
Focus Follows Mouse

### DISCUSSION THREAD

**Discussion by wesbluemarine** [YaLTeR/niri#47] [Ideas]
Could it be possible to change to this behaviour?  It would become a mouse based only lateral scrolling.
Thanks for your work!

**Comment by YaLTeR** [comment]
Hmm, not sure it's the best idea. Imagine with two 67% windows moving your mouse around the screen causing wild scrolling left and right.

**Reply by wesbluemarine** [reply]
Yes, It could be a mess. Maybe could be better to implement an option to enable focus follows mouse but without auto scrolling.
So you can interact with partially visible windows without scrolling. 
Auto scrolling would be enabled only on window click.
What do you think?

EDIT
I only realized now that niri is already like this! 
By moving on a window without clicking it is possible to interact.

**Reply by YaLTeR** [reply]
Sorry, apparently GitHub did not subscribe me to discussion answers. But yeah, you don't need to focus a window to scroll it or such. This is true for other compositors too. Only for keyboard input you need to focus.

**Reply by Connor-GH** [reply]
I personally use niri (yes I am from the Matrix room) and I come from Hyprland. The click feels not needed.

**Reply by phuhl** [reply]
> EDIT
> I only realized now that niri is already like this!
> By moving on a window without clicking it is possible to interact.

Is it? For me, moving the mouse over a window (w/o clicking) does not change where keystrokes go... Would be great to have, though. The `focus-follows-mouse` is a bit too extreme for me, with the auto scrolling. On a multi monitor setup, I tend to get lost with it, so having a `keyboard-focus-follows-mouse` would be a great addition for me.

**Reply by YaLTeR** [reply]
> does not change where **keystrokes** go

Yeah, keyboard focus does not change. What I meant was pointer focus, i.e. you can use the mouse wheel over unfocused windows.

> The focus-follows-mouse is a bit too extreme for me, with the auto scrolling. On a multi monitor setup, I tend to get lost with it, so having a keyboard-focus-follows-mouse would be a great addition for me.

I understand your problem, but I don't think keyboard-focus-follows-mouse is the right solution. The current behavior flows naturally from a combination of factors:
1. mouse movement causes a different window to get (keyboard) focus
2. focused window scrolls into view

A better way might be some way to freeze scrolling so that focused window doesn't scroll into view immediately.

**Reply by phuhl** [reply]
> A better way might be some way to freeze scrolling so that focused window doesn't scroll into view immediately.

Yeah, sounds good, too. Would be great, especially for password managers and the like. Everywhere, where you need to copy and past stuff by quickly by mouse and shortcut.

**Comment by YaLTeR** [comment]
Implemented this in df9d721f7462a7edaacb9de4530580a68ab6e8ab
